src/coreclr/nativeaot/Runtime/unix/UnixNativeCodeManager.cpp

```diff
@@ -722,6 +722,63 @@ int UnixNativeCodeManager::IsInProlog(MethodInfo * pMethodInfo, PTR_VOID pvAddre
 
     return savedFp && savedRa && establishedFp ? 0 : 1;
 
+#elif defined(TARGET_RISCV64)
+
+// store pair with signed offset
+// 0100 00xx xxxxxxxx xxxx xxxx xxxx xxxx
+#define STW_PAIR_BITS 0x04000000
+#define STW_PAIR_MASK 0xFC000000
+
+// add fp, sp, x
+// addi fp, sp, x
+// 0000 0001 100x xxxx xxxx xxxx 0000 0000
+#define ADD_FP_SP_BITS 0x01C00000
+#define ADD_FP_SP_MASK 0xFFFFE000
+
+#define STW_PAIR_RS1_MASK 0xF80
+#define STW_PAIR_RS1_SP  0xF80
+#define STW_PAIR_RS1_FP  0xF00
+#define STW_PAIR_RS2_MASK 0xF00
+#define STW_PAIR_RS2_FP  0xF00
+#define STW_PAIR_RS2_RA  0xF40
+
+    UnixNativeMethodInfo * pNativeMethodInfo = (UnixNativeMethodInfo *)pMethodInfo;
+    ASSERT(pNativeMethodInfo != NULL);
+
+    uint32_t* start  = (uint32_t*)pNativeMethodInfo->pMethodStartAddress;
+    bool savedFp = false;
+    bool savedRa = false;
+    bool establishedFp = false;
+
+    for (uint32_t* pInstr = (uint32_t*)start; pInstr < pvAddress && !(savedFp && savedRa && establishedFp); pInstr++)
+    {
+        uint32_t instr = *pInstr;
+
+        if (((instr & STW_PAIR_MASK) == STW_PAIR_BITS) &&
```

Comment:
Should be something like TrailingEpilogueInstructionsCount but detecting stores instead of loads.